lib/firmware-base/system-status.c

```diff
@@ -0,0 +1,94 @@
+#include <driver/timer.h>
+#include <freertos/FreeRTOS.h>
+#include <freertos/timers.h>
+
+#include <stdio.h>
+
+#define TIMER_DIVIDER 80  // Divide Base by 80
+
+
+// Timer group 0, timer 0 configuration
+timer_config_t timer_config = {
+	.alarm_en    = TIMER_ALARM_DIS,	     // Disable alarm
+	.intr_type   = TIMER_INTR_DISABLE,   // No timer interrupts
+	.counter_dir = TIMER_COUNT_UP,	     // Count up
+	.auto_reload = TIMER_AUTORELOAD_EN,  // Auto-reload
+	.divider = TIMER_DIVIDER  // Set divider for 1ms (assumption that base
```

Comment:
If you do expand this into a proper `#define` you wouldn't need to make this assumption! FreeRTOS should have a config `#define` that has the CPU frequency.